runelite-client/src/main/java/net/runelite/client/plugins/camera/CameraPlugin.java

```diff
@@ -117,36 +128,84 @@ protected void shutDown()
 		controlDown = false;
 	}
 
+	@Subscribe
+	public void onGameTick(GameTick gameTick)
+	{
+		final int timeoutTicks = (config.compassMultipleClickTimeout() * 1000) / Constants.GAME_TICK_LENGTH;
+		if (currentLeftClickOption.equals(defaultLeftClickOption) || (client.getTickCount() < (lastUpdatedTick + timeoutTicks)))
+		{
+			return;
+		}
+		currentLeftClickOption = defaultLeftClickOption;
+	}
+
 	@Subscribe
 	public void onMenuEntryAdded(MenuEntryAdded menuEntryAdded)
 	{
-		if (menuEntryAdded.getType() == MenuAction.CC_OP.getId() && menuEntryAdded.getOption().equals(LOOK_NORTH) && config.compassLook())
+		if (menuEntryAdded.getType() == MenuAction.CC_OP.getId())
 		{
-			MenuEntry[] menuEntries = client.getMenuEntries();
-			int len = menuEntries.length;
-			MenuEntry north = menuEntries[len - 1];
+			if (ArrayUtils.indexOf(options, menuEntryAdded.getOption()) < 0)
+			{
+				return;
+			}
 
-			menuEntries = Arrays.copyOf(menuEntries, len + 3);
+			final int count = options.length;
+			MenuEntry[] menuEntries = client.getMenuEntries();
+			int currentOptionIndex = ArrayUtils.indexOf(options, currentLeftClickOption);
 
-			// The handling for these entries is done in ToplevelCompassOp.rs2asm
-			menuEntries[--len] = createCameraLookEntry(menuEntryAdded, 4, LOOK_WEST);
-			menuEntries[++len] = createCameraLookEntry(menuEntryAdded, 3, LOOK_EAST);
-			menuEntries[++len] = createCameraLookEntry(menuEntryAdded, 2, LOOK_SOUTH);
-			menuEntries[++len] = north;
+			if (config.compassLook())
+			{
+				menuEntries = Arrays.copyOf(menuEntries, menuEntries.length + 3);
+
+				// fill the menu from top to bottom
+				// but iterate through options cyclically from 0 -> 3 -> 0
+				int entryIndex = menuEntries.length - 1;
+
+				// The handling for these entries is done in ToplevelCompassOp.rs2asm`
+				menuEntries[entryIndex] = createCameraLookEntry(menuEntryAdded, currentOptionIndex + 1, options[currentOptionIndex % count]);
+				entryIndex -= 1;
+				currentOptionIndex += 1;
+				menuEntries[entryIndex] = createCameraLookEntry(menuEntryAdded, currentOptionIndex + 1, options[currentOptionIndex % count]);
+				entryIndex -= 1;
+				currentOptionIndex += 1;
+				menuEntries[entryIndex] = createCameraLookEntry(menuEntryAdded, currentOptionIndex + 1, options[currentOptionIndex % count]);
+				entryIndex -= 1;
+				currentOptionIndex += 1;
+				menuEntries[entryIndex] = createCameraLookEntry(menuEntryAdded, currentOptionIndex + 1, options[currentOptionIndex % count]);
+			}
+			else
+			{
+				// set top entry to the current left click
+				menuEntries[menuEntries.length - 1] = createCameraLookEntry(menuEntryAdded, currentOptionIndex + 1, options[currentOptionIndex % count]);
```

Comment:
I don't understand... Why is `onMenuEntryAdded` running any code when `config.compassLook()` is false? We shouldn't be adding or changing any menu entries when that's the case.

Reply:
if `config.compassLook() == true` then we need to build out the menu options for that plugin, and rearrange them depending on the current left click option. If `config.compassLook() == false` then we still need to modify the top option to Look South, East, West, etc depending on the current left click value. 

If `config.compassMultipleClickTimeout() == false` (which is the config option I'm adding) then we just don't do anything special on a compass click, instead `currentLeftClickOption` will always be the default "Look North"